src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,38 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        turnX(robot, toX);
+        while (robot.getX() != toX) {
+            robot.stepForward();
+        }
+        turnY(robot, toY);
+        while (robot.getY() != toY) {
+            robot.stepForward();
+        }
+    }
+
+    public void turnX(Robot robot, int toX) {
+        if (robot.getX() > toX) {
+            while (robot.getDirection() != Direction.LEFT) {
+                robot.turnRight();
+            }
+        } else {
+            while (robot.getDirection() != Direction.RIGHT) {
+                robot.turnRight();
+            }
+        }
+    }
+
+    public void turnY(Robot robot, int toY) {
+        if (robot.getY() > toY) {
+            while (robot.getDirection() != Direction.DOWN) {
+                robot.turnRight();
+            }
+        } else {
+            while (robot.getDirection() != Direction.UP) {
+                robot.turnRight();
+            }
+        }
     }
```

Comment:
What is the difference in logic between these methods?

Reply:
I thought about it yesterday but could not find a solution so I sent it. This is new solution.